src/tensornetwork.jl

```diff
@@ -0,0 +1,91 @@
+using Dictionaries: Dictionary
+using Graphs: Graphs
+using ITensors: ITensors, ITensor
+using NamedGraphs: NamedGraphs, add_edge!, incident_edges
+using NamedGraphs.GraphsExtensions: rem_edges!
+using Adapt
+
+#TODO: Make this show() nicely.
```

Comment:
I would move the `abstracttensornetwork.jl` `tensornetwork.jl` and `tensornetworkstate.jl` files into a directory.